lumicks/pylake/correlated_stack.py

```diff
@@ -8,9 +8,52 @@
 from .adjustments import ColorAdjustment
 from .detail.image import make_image_title
 from .detail.imaging_mixins import FrameIndex, TiffExport, VideoExport
+from .detail.plotting import get_axes, show_image
 from .detail.widefield import TiffStack
 
 
+def _deprecate_plot_arguments(plot):
+    """Decorator to deprecate old arguments of the method `CorrelatedStack.plot()`"""
+    import functools
+    import matplotlib
+    import warnings
+
+    def is_old_order(args):
+        # old arguments: frame, channel, show_title, axes, adjustment
+        old_arg_types = [int, str, bool, matplotlib.axes.Axes, ColorAdjustment]
+        nones_allowed = [False, True, False, True, False]
+        old_order = 1 <= len(args) <= 5
+        for arg, old_type, none_allowed in zip(args, old_arg_types, nones_allowed):
+            old_order = old_order and (isinstance(arg, old_type) or none_allowed and arg is None)
+        return old_order
+
+    @functools.wraps(plot)
+    def wrapper(self, *args, **kwargs):
+        # The plot function might be called with up to 5 positional arguments with the old order
+        # We can gracefully convert the old ordered positional arguments into keyword arguments
+        if is_old_order(args):
+            keys = ["frame", "channel", "show_title", "axes", "adjustment"]
+            defaults = [0, "rgb", True, None, ColorAdjustment.nothing()]
+            for arg, key, default in zip(args, keys, defaults):
+                if key in kwargs:
+                    raise TypeError(
+                        f"`{self.__class__.__name__}.plot()` got multiple values for argument `{key}`"
+                    )
+                kwargs[key] = default if arg is None else arg
+            warnings.warn(
+                DeprecationWarning(
+                    f"The call signature of `plot()` has changed: Please, provide `{key}` as a "
+                    "keyword argument."
+                ),
+                stacklevel=2,
+            )
+            return plot(self, **kwargs)
+        else:
+            return plot(self, *args, **kwargs)
```

Comment:
There's a second failure mode that is maybe a bit more notorious which is that:

```python
cs.plot("red", 3)
```
goes here and complains that there are too many arguments. Unavoidable I guess. Not a big deal though, since even in the deprecated order, this was the wrong argument order.

Reply:
Yeah, I noticed that, too, but accepted it with the same reasoning you mentioned.